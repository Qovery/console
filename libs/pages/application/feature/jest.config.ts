/* eslint-disable */
export default {
  displayName: 'pages-application-feature',
  preset: '../../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/pages/application/feature',
}
