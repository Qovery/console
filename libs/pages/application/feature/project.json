{
  "root": "libs/pages/application/feature",
  "sourceRoot": "libs/pages/application/feature/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/pages/application/feature/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/pages/application/feature"],
      "options": {
        "jestConfig": "libs/pages/application/feature/jest.config.js",
        "passWithNoTests": true
      }
    }
  }
}
