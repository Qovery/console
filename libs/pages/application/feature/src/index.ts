export * from './lib/application-page'
