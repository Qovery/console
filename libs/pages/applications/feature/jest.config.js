module.exports = {
  displayName: 'pages-applications-feature',
  preset: '../../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/pages/applications/feature',
}
