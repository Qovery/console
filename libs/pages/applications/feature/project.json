{
  "root": "libs/pages/applications/feature",
  "sourceRoot": "libs/pages/applications/feature/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/pages/applications/feature/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/pages/applications/feature"],
      "options": {
        "jestConfig": "libs/pages/applications/feature/jest.config.js",
        "passWithNoTests": true
      }
    }
  }
}
