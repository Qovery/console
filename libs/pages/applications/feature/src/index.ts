export * from './lib/applications-page'
export * from './lib/components/general/general'
