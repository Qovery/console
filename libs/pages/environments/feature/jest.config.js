module.exports = {
  displayName: 'pages-environments-feature',
  preset: '../../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/pages/environments/feature',
}
