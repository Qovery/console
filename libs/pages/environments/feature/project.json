{
  "root": "libs/pages/environments/feature",
  "sourceRoot": "libs/pages/environments/feature/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/pages/environments/feature/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/pages/environments/feature"],
      "options": {
        "jestConfig": "libs/pages/environments/feature/jest.config.js",
        "passWithNoTests": true
      }
    }
  }
}
