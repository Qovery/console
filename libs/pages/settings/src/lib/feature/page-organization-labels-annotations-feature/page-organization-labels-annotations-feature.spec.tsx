import * as organizationsDomain from '@qovery/domains/organizations/feature'
import * as utilDates from '@qovery/shared/util-dates'
import { renderWithProviders } from '@qovery/shared/util-tests'
import PageOrganizationLabelsAnnotationsFeature from './page-organization-labels-annotations-feature'

const useAnnotationsGroupsMockSpy = jest.spyOn(organizationsDomain, 'useAnnotationsGroups') as jest.Mock
const useLabelsGroupsMockSpy = jest.spyOn(organizationsDomain, 'useLabelsGroups') as jest.Mock

const mockAnnotationsGroup = [
  {
    id: '0',
    created_at: '2024-04-08T18:51:39.836359Z',
    updated_at: '2024-04-09T08:05:09.470801Z',
    name: 'test',
    annotations: [
      {
        key: 'key',
        value: 'value',
      },
    ],
    scopes: ['PERSISTENT_VOLUME_CLAIMS', 'REPLICA_SETS', 'INGRESS'],
  },
]

const mockLabelsGroup = [
  {
    id: '0',
    created_at: '2024-04-08T18:51:39.836359Z',
    updated_at: '2024-04-09T08:05:09.470801Z',
    name: 'test',
    labels: [
      {
        key: 'key',
        value: 'value',
        propagate_to_cloud_provider: false,
      },
    ],
  },
]

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({ organizationId: '1' }),
}))

describe('PageOrganizationLabelsAnnotationsFeature', () => {
  beforeEach(() => {
    useAnnotationsGroupsMockSpy.mockReturnValue({
      data: mockAnnotationsGroup,
      isFetched: true,
    })
    useLabelsGroupsMockSpy.mockReturnValue({
      data: mockLabelsGroup,
      isFetched: true,
    })
  })

  it('should match snapshot', () => {
    jest.spyOn(utilDates, 'timeAgo').mockReturnValue('1 month')

    const { baseElement } = renderWithProviders(<PageOrganizationLabelsAnnotationsFeature />)
    expect(baseElement).toMatchSnapshot()
  })
})
