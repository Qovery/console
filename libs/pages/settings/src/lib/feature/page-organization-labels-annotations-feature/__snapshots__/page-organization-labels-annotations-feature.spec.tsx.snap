// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageOrganizationLabelsAnnotationsFeature should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex w-full flex-col justify-between"
    >
      <section
        class="flex flex-col max-w-content-with-navigation-left p-8"
      >
        <div
          class="mb-8 flex justify-between gap-3"
        >
          <div
            class="space-y-3"
          >
            <h1
              class="text-neutral-400 text-xl font-bold"
            >
              Labels & Annotations
            </h1>
            <p
              class="text-xs text-neutral-400"
            >
              Define and manage the labels & annotations to be used within your organization.
            </p>
            <button
              class="inline-flex shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-xs h-6 px-2 rounded bg-brand-50 border border-brand-500 text-brand-500 hover:[&:not(:active):not(:disabled)]:bg-white w-fit items-center gap-1"
            >
              <i
                aria-hidden="true"
                class="fa-regular fa-circle-question p-0.5 text-xs"
              />
              <span>
                Need help here?
              </span>
            </button>
          </div>
          <button
            aria-expanded="false"
            aria-haspopup="menu"
            class="inline-flex items-center font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-9 px-3 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100 shrink-0 gap-2"
            data-state="closed"
            id="radix-:r6:"
            type="button"
          >
            Add new
            <i
              aria-hidden="true"
              class="fa-regular fa-circle-plus "
            />
          </button>
        </div>
        <div
          class="mb-5 rounded border border-neutral-250 bg-neutral-100 w-full"
          data-testid="block-content"
        >
          <div
            class="flex h-9 items-center justify-between border-b border-neutral-250 px-4"
          >
            <h2
              class="text-ssm font-medium text-neutral-400"
            >
              Labels groups
            </h2>
          </div>
          <div
            class="p-0"
          >
            <ul>
              <li
                class="flex items-center justify-between border-b border-neutral-250 px-5 py-4 last:border-0"
              >
                <div
                  class="flex flex-col"
                >
                  <h2
                    class="mb-1 flex text-xs font-medium text-neutral-400"
                  >
                    test
                  </h2>
                  <div
                    class="flex text-xs text-neutral-350"
                  >
                    <p>
                      Created:
                       
                      <span
                        class="text-neutral-400"
                      >
                        Apr 8, 2024
                      </span>
                    </p>
                    <p
                      class="ml-3"
                    >
                      Updated: 
                      <span
                        class="text-neutral-400"
                      >
                        1 month
                      </span>
                    </p>
                  </div>
                </div>
                <div
                  class="flex gap-2"
                >
                  <button
                    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400 relative mr-2"
                  >
                    <span
                      class="absolute -right-1 -top-1 flex h-3 w-3 items-center justify-center rounded-full bg-brand-500 text-3xs font-bold leading-[0] text-white"
                    />
                    <i
                      aria-hidden="true"
                      class="fa-regular fa-layer-group "
                    />
                  </button>
                  <button
                    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
                  >
                    <i
                      aria-hidden="true"
                      class="fa-regular fa-gear "
                    />
                  </button>
                  <button
                    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
                    data-state="closed"
                    disabled=""
                  >
                    <i
                      aria-hidden="true"
                      class="fa-regular fa-trash-can "
                    />
                  </button>
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="mb-5 rounded border border-neutral-250 bg-neutral-100 w-full"
          data-testid="block-content"
        >
          <div
            class="flex h-9 items-center justify-between border-b border-neutral-250 px-4"
          >
            <h2
              class="text-ssm font-medium text-neutral-400"
            >
              Annotations groups
            </h2>
          </div>
          <div
            class="p-0"
          >
            <ul>
              <li
                class="flex items-center justify-between border-b border-neutral-250 px-5 py-4 last:border-0"
              >
                <div
                  class="flex flex-col"
                >
                  <h2
                    class="mb-1 flex text-xs font-medium text-neutral-400"
                  >
                    test
                  </h2>
                  <div
                    class="flex text-xs text-neutral-350"
                  >
                    <p>
                      Created:
                       
                      <span
                        class="text-neutral-400"
                      >
                        Apr 8, 2024
                      </span>
                    </p>
                    <p
                      class="ml-3"
                    >
                      Updated:
                       
                      <span
                        class="text-neutral-400"
                      >
                        1 month
                      </span>
                    </p>
                  </div>
                </div>
                <div
                  class="flex gap-2"
                >
                  <button
                    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400 relative mr-2"
                  >
                    <span
                      class="absolute -right-1 -top-1 flex h-3 w-3 items-center justify-center rounded-full bg-brand-500 text-3xs font-bold leading-[0] text-white"
                    />
                    <i
                      aria-hidden="true"
                      class="fa-regular fa-layer-group "
                    />
                  </button>
                  <button
                    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
                  >
                    <i
                      aria-hidden="true"
                      class="fa-regular fa-gear "
                    />
                  </button>
                  <button
                    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
                    data-state="closed"
                    disabled=""
                  >
                    <i
                      aria-hidden="true"
                      class="fa-regular fa-trash-can "
                    />
                  </button>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </div>
</body>
`;
